### PR TITLE
PWM: Fixed frequency prescalers, removed calloc on init

### PR DESCRIPTION
**Pull request by vsyn**
The 15th bit in the compare register inverts the polarity of the output, therefore the prescalers needed adjusting to only use a maximum of 15 bits for the max value.

Removed the calloc on init too.

### CODE REVIEW

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ struct nrf53_pwm_dev_global {
     bool playing;
     nrf_drv_pwm_t drv_instance;
     nrf_drv_pwm_config_t config;
-    nrf_pwm_values_individual_t *duty_cycles;
+    nrf_pwm_values_individual_t duty_cycles[4];
```
Comment (mkiiskila) [inline hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: Should this be NRF_PWM_CHANNEL_COUNT, instead of 4?

Comment (vsyn) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: Fixed, thanks.

**Comment by mlaz** [comment]
> The 15th bit in the compare register inverts the polarity of the output
@vsyn Is this documented anywhere you can post a link to?

**Comment by vsyn** [comment]
> Is this documented anywhere you can post a link to?

[nRF52 product specification](http://infocenter.nordicsemi.com/pdf/nRF52832_PS_v1.1.pdf) page 498 2nd paragraph

**Pull request merged**